user_guide_src/source/database/query_builder/115.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+$query = $this->db->table('user2')
+    ->select('user2.name, user2.email, user2.country')
+    ->join('user', 'user.email = user2.email', 'left')
+    ->where('user.email IS NULL');
+
+$additionalUpdateField = ['updated_at' => new RawSql('CURRENT_TIMESTAMP')];
+
+$sql = $builder->onConstraint('email')->updateFields($additionalUpdateField, true)->upsertBatch($query);
```

Comment:
I'm not comfortable with this code `upsertBatch($query)`.
I feel that the following way is sufficient.

```php
$sql = $builder->setQueryAsData($query)->onConstraint('email')
        ->updateFields($additionalUpdateField, true)->upsertBatch();
```

Do we really need to accept a query with `*batch()` methods?

Reply:
I was thinking we might should leave this out too. I went ahead and removed it.